pkg/operators/ustack/ustack.go

```diff
@@ -0,0 +1,247 @@
+// Copyright 2025 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package ustack provides a data operator that resolves user stack traces.
+package ustack
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/cilium/ebpf"
+
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/datasource"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadget-service/api"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/operators"
+	ebpfoperator "github.com/inspektor-gadget/inspektor-gadget/pkg/operators/ebpf"
+	ebpftypes "github.com/inspektor-gadget/inspektor-gadget/pkg/operators/ebpf/types"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/params"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/symbolizer"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/utils/annotations"
+)
+
+const (
+	Name     = "ustack"
+	Priority = 100
+
+	PerfMaxStackDepth = 127
+)
+
+const (
+	userStackTargetNameAnnotation = "ebpf.formatter.ustack"
+)
+
+type Operator struct{}
+
+func (o *Operator) Name() string {
+	return Name
+}
+
+func (o *Operator) Init(params *params.Params) error {
+	return nil
+}
+
+func (o *Operator) GlobalParams() api.Params {
+	return nil
+}
+
+func (o *Operator) InstanceParams() api.Params {
+	return nil
+}
+
+func (o *Operator) InstantiateDataOperator(gadgetCtx operators.GadgetContext, instanceParamValues api.ParamValues) (operators.DataOperatorInstance, error) {
+	instance := &OperatorInstance{
+		symbolizer:    symbolizer.NewSymbolizer(),
+		subscriptions: make(map[datasource.DataSource][]func(ds datasource.DataSource, data datasource.Data) error),
+	}
+	err := instance.init(gadgetCtx)
+	if err != nil {
+		return nil, err
+	}
+	return instance, nil
+}
+
+func (o *Operator) Priority() int {
+	return Priority
+}
+
+type OperatorInstance struct {
+	userStackMap *ebpf.Map
+	symbolizer   *symbolizer.Symbolizer
+
+	subscriptions map[datasource.DataSource][]func(ds datasource.DataSource, data datasource.Data) error
+}
+
+func (o *OperatorInstance) init(gadgetCtx operators.GadgetContext) error {
+	logger := gadgetCtx.Logger()
+
+	for _, ds := range gadgetCtx.GetDataSources() {
+		for _, in := range ds.GetFieldsWithTag("type:" + ebpftypes.UserStackTypeName) {
+			if in == nil {
+				continue
+			}
+			in.SetHidden(true, false)
```

Comment:
```suggestion
			in.SetHidden(true, true)
```

Otherwise I get:

```bash 
$ sudo ig run trace_capabilities:alban_user_stack --verify-image=false --print-ustack
WARN[0001] image signature verification is disabled due to using corresponding option
WARN[0001] image signature verification is disabled due to using corresponding option
RUNTIME.CONTAINERN… COMM                PID        TID CAPA… AUDIT  USTACK_RA… USTACK_RA… USTACK_RA… USTACK_RA… CAP        SYSCALL
```